mudata/_core/mudata.py

```diff
@@ -555,6 +668,14 @@ def _update_attr(self, attr: str, axis: int, join_common: bool = False):
                 if mx_key not in self.mod.keys():  # not a modality name
                     attrp[mx_key] = attrp[mx_key][keep_index, keep_index]
 
+            # Write ._mod_index
+            # if attr_names_maybe_changed:
```

Comment:
Is this supposed to be commented out? You don't write to `self._mod_index` anywhere else

Reply:
As mentioned in https://github.com/scverse/mudata/issues/18, there's no clarity if and how `._mod_index` will speed things up and should be implemented.
More generally, if the index hasn't changed, I envision we can just copy columns accordingly using existing `.attrmap` avoiding joins, but that would require quite a separate «branch» of code handling this logic. (We won't be able to re-use the index in joins when they are non-unique values even if it hasn't changed.)

If you have thoughts on this, please contribute to #18!